src/tree/settings/CommonProperties.ts

```diff
@@ -52,3 +53,28 @@ export function getCommonGroupIcon(property: CommonProperty | CommonGroupBy): Th
 
     return new ThemeIcon(icon);
 }
+
+export const NonLabelGroupName = l10n.t('others');
+
+export function getLabel(containerLabels: Labels, itemType: 'containers' | 'images'): string {
+    if (!containerLabels) {
+        return NonLabelGroupName;
+    }
+
+    const labelName = workspace.getConfiguration('docker')?.get<string | undefined>(`${itemType}.groupByLabel`, undefined);
+    if (!labelName) {
+        return NonLabelGroupName;
+    }
+
+    const labels = Object.keys(containerLabels).map(label => ({ label: label, value: containerLabels[label] }));
```

Comment:
JS lets you directly index map objects, since they're ultimately just regular old objects. So you can shorten this block + the rest of the method body--

```typescript
return containerLabels?.[labelName] || NonLabelGroupName;
```
Essentially what that line means is, from the `containerLabels` object, if it has a property with name matching `labelName`'s value, then it will return that--or, if it doesn't, or if the value is null, undefined, or falsy, then it will return `NonLabelGroupName`'s value instead.

Reply:
That simplifies the method by a lot, thanks Brandon :)

Reply:
I think these lines need to stay, before that return:
```typescript
    if (!labelName) {
        return NonLabelGroupName;
    }
```

Reply:
(_Technically_ they don't, as `item?.[undefined]` evaluates to `undefined` which would be falsy, so falling into that `NonLabelGroupName`, but that's a sort of esoteric behavior, so keeping these lines is better for clarity)

Reply:
I agree, that is a bit confusing. I didn't think about that at all.